pkg/kubelet/kubelet_pods.go

```diff
@@ -130,6 +132,17 @@ func (kl *Kubelet) getKubeletMappings() (uint32, uint32, error) {
 
 	if !utilfeature.DefaultFeatureGate.Enabled(features.UserNamespacesSupport) {
 		return defaultFirstID, defaultLen, nil
+	} else {
+		kernelVersion, err := utilkernel.GetVersion()
+		if err != nil {
+			return 0, 0, fmt.Errorf("failed to get kernel version, unable to determine if feature %s can be supported : %w",
+				features.UserNamespacesSupport, err)
+		}
+		if kernelVersion != nil && !kernelVersion.AtLeast(version.MustParseGeneric(utilkernel.UserNamespacesSupportKernelVersion)) {
+			return 0, 0, fmt.Errorf(
+				"the kernel version (%s) is incompatible with the %s feature gate, which needs %s as a minimum kernel version",
+				kernelVersion, features.UserNamespacesSupport, utilkernel.UserNamespacesSupportKernelVersion)
+		}
```

Comment:
This is breaking the kind job that monitors the state of a cluster with alpha features enabled 

https://testgrid.k8s.io/sig-testing-kind#kind-master-alpha-beta

The kubelet fails to start https://storage.googleapis.com/kubernetes-jenkins/logs/ci-kubernetes-e2e-kind-alpha-beta-features/1789720580090499072/artifacts/kind-control-plane/kubelet.log

If I understand the feature correctly, the enablement of the feature gate does not necessary means that kernel version is required.
It seems the feature gate is being used as a flag that is incorrect, as if this graduates it will be always enabled, and we can not require `UserNamespacesSupportKernelVersion` for ALL the cluster

@dims @AkihiroSuda can you please relax this validation to check the kernel version supported ONLY when the feature is actually used? 

/cc @liggitt